### PR TITLE
Add private key onboarding flow

### PR DESCRIPTION
**Pull request by jagodarybacka**
Resolves https://github.com/tahowallet/extension/issues/3069

### What

This PR adds private key flow to tabbed onboarding and allows wallet to show these wallets in the UI correctly

- make seed phrase safe input a separate, shared component
- refactor seed phrase form to make it reusable
- add new page for private key import
- show private key imported wallets in a new section on the accounts list

https://user-images.githubusercontent.com/20949277/223729085-1c5c2c18-3f4d-4fb9-8dbc-0193c458f902.mov

image

### Testing

1. Go through the private key onboarding flow
2. Import account with private key
3. Check how it behaves in the wallet
4. Import other accounts, check wallets list - private key wallets should be in `Generated and imported` section

Latest build: [extension-builds-3119](https://github.com/tahowallet/extension/suites/12013923000/artifacts/631100120) (as of Tue, 04 Apr 2023 09:23:44 GMT).

### CODE REVIEW

**Comment by jagodarybacka** [comment]
Outstanding issue - private key input validation
 For mnemonic, we are only checking the length. What rules do we want to include here? Check if it is a valid hex string? Check length?

**Comment by VladUXUI** [comment]
>Check if it is a valid hex string? Check length?

Can we do both?

File: background/redux-slices/selectors/signingSelectors.ts [hunk 1]
```diff
@@ -62,15 +69,36 @@ export const selectAccountSignersByAddress = createSelector(
       )
       .filter(isDefined)
 
+    const privateKeyEntires = Object.entries(walletsByAddress)
```
Comment (0xDaedalus) [inline background/redux-slices/selectors/signingSelectors.ts:72]:
```suggestion
    const privateKeyEntries = Object.entries(walletsByAddress)
```

File: ui/_locales/en/messages.json [hunk 2]
```diff
@@ -350,6 +352,13 @@
             "invalidPhraseError": "Invalid recovery phrase"
           }
         },
+        "importPrivateKey": {
+          "title": "Import private key",
+          "subtitle": "Importing a private key does not associate it to any secret recovery phrase, but it’s still protected by the same password",
```
Comment (0xDaedalus) [inline ui/_locales/en/messages.json:357]:
```suggestion
          "subtitle": "Importing a private key does not associate it to a secret recovery phrase, but it’s still protected by the same password",
```

File: ui/components/AccountsNotificationPanel/AccountsNotificationPanelAccounts.tsx [hunk 3]
```diff
@@ -51,6 +51,14 @@ type WalletTypeInfo = {
   category: string
 }
 
+const accountTypes = [
```
Comment (0xDaedalus) [inline ui/components/AccountsNotificationPanel/AccountsNotificationPanelAccounts.tsx:54]: I think you can do `Object.values(AccountType)` here

File: ui/components/AccountsNotificationPanel/AccountsNotificationPanelAccounts.tsx [hunk 4]
```diff
@@ -290,145 +328,135 @@ export default function AccountsNotificationPanelAccounts({
     }
   }, [history, areKeyringsUnlocked, dispatch, t])
 
-  const accountTypes = [
-    AccountType.Internal,
-    AccountType.Imported,
-    AccountType.ReadOnly,
-    AccountType.Ledger,
-  ]
+  const existingAccountTypes = accountTypes.filter(
```
Comment (0xDaedalus) [inline ui/components/AccountsNotificationPanel/AccountsNotificationPanelAccounts.tsx:331]: 🔥

**Review by 0xDaedalus — APPROVED** [review]
LGTM - feel free to merge and address comments in a separate PR - or add a commit here and then merge.

File: ui/components/AccountsNotificationPanel/AccountsNotificationPanelAccounts.tsx (hunk 3 above)
Comment (jagodarybacka) [reply ui/components/AccountsNotificationPanel/AccountsNotificationPanelAccounts.tsx:54]: Typescript is not happy to iterate over enum `AccountType` like that 😞

**Pull request merged**